### PR TITLE
normalize version for vendors without minor/patch version

### PR DESCRIPTION
**Pull request by roele**
Updates the scripts of vendors which ship major versions without minor/patch to normalize the version.

**Vendors**
* liberica
* openjdk
* oracle
* sapmachine

### CODE REVIEW

**Comment by roele** [comment]
Tested this shortly with the following`/bin/test.bash` file.

```bash
#!/usr/bin/env bash
set -e
set -Euo pipefail

# shellcheck source=bin/functions.bash
source "$(dirname "${0}")/functions.bash"

IFS=$'\r\n' GLOBIGNORE='*' command eval  'versions=($(mise ls-remote java))'

for version in "${versions[@]}" 
do
  # does not start with openjdk-
  # if [[ "${version}" != openjdk-* ]]; then
  #   continue
  # fi
  echo "$(normalize_version "${version}")"
done
```

Only version which is not accounted for is `openjdk-23-loom+2-48.0.0` for which i don't necessarily want a specific regex.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -11,9 +11,15 @@ jobs:
     runs-on: 'ubuntu-latest'
     steps:
     - uses: actions/checkout@v4
+      with:
+        submodules: 'true'
     - name: Install dependencies
-      run: sudo apt -y install shellcheck
+      run: |
+        apt-get update
+        apt -y install shellcheck
```
Comment (jdx) [inline .github/workflows/test.yml:19]:
```suggestion
      run: sudo apt-get update && apt-get install -y shellcheck
```

though really we should be using mise-action :)

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -11,9 +11,13 @@ jobs:
     runs-on: 'ubuntu-latest'
     steps:
     - uses: actions/checkout@v4
+      with:
+        submodules: 'true'
     - name: Install dependencies
-      run: sudo apt -y install shellcheck
+      run: sudo apt-get update && apt-get install -y shellcheck
```
Comment (jdx) [inline .github/workflows/test.yml:17]:
```suggestion
      run: sudo apt-get update && sudo apt-get install -y shellcheck
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (roele) [reply .github/workflows/test.yml]: agreed, we should switch to mise actions at some point

**Pull request merged**